frontend/event/eventController.js

```diff
@@ -322,30 +332,44 @@
         }
 
         function addLinks(event) {
-            let videoUrls = _.filter(dialogCtrl.videoUrls, videoUrl => videoUrl.url !== '');
-            let usefulLinks = _.filter(dialogCtrl.usefulLinks, usefulLink => usefulLink.url !== '');
+            let videoUrls = dialogCtrl.videoUrls.filter(videoUrl => videoUrl.url !== '');
+            let usefulLinks = dialogCtrl.usefulLinks.filter(usefulLink => usefulLink.url !== '');
             event.video_url = videoUrls;
             event.useful_links = usefulLinks;
         };
 
         function getCountries() {
             $http.get('app/institution/countries.json').then(function success(response) {
                 dialogCtrl.countries = response.data;
-                dialogCtrl.event.country = "Brasil";
             });
         }
 
+        function loadSelectedState() {
+            if (dialogCtrl.event.address.federal_state) {
+                dialogCtrl.selectedFederalState = dialogCtrl.federalStates
+                    .filter(federalState => federalState.nome === dialogCtrl.event.address.federal_state)
+                    .reduce(federalState => federalState);
+                dialogCtrl.getCitiesByState();
+           }
+        }
+
         (function main() {
             getCountries();
             loadFederalStates();
             if(dialogCtrl.event) {
                 dialogCtrl.dateChangeEvent = _.clone(dialogCtrl.event);
                 dialogCtrl.dateChangeEvent.start_time = new Date(dialogCtrl.dateChangeEvent.start_time);
                 dialogCtrl.dateChangeEvent.end_time = new Date(dialogCtrl.dateChangeEvent.end_time);
+                dialogCtrl.start_time = new Date(dialogCtrl.dateChangeEvent.start_time);
```

Comment:
Try to think how to solve the problem to one date and generalize into a single function, pull out those dates to a data structure and iterate over solving the problem.